h/static/scripts/annotator/plugin/enhancedanchoring.coffee

```diff
@@ -83,6 +89,37 @@ class Anchor
       # Kill the list if it's empty
       delete @anchoring.anchors[index] unless anchors.length
 
+  # Scroll to this anchor
+  scrollIntoView: ->
```

Comment:
> Do you think this should be a method on the Anchor or would it be more useful for the Anchor to expose a commonAncestor Element or something like this? I'm just wondering because scrollIntoView is an Element API and rather than bubbling that API up to the Anchor maybe we should just expose an underlying Element.

It's more complicated than that. First, in this context, "scrolling" to an anchor might not actually mean physically scrolling a DOM element. It can involve completely different things, like calling arbitrary APIs of the underlying platform to access a specific location in the document. Furthermore, it's also possible that the Element (that we want to scroll to) might not even exist at this point. (If the anchor has not yet been realized, there is no highlight in the DOM, and maybe no relevant DOM at all.)

That's exactly why we have this API here. This wraps everything: do whatever we need to do in order to have the relevant part of the document rendered; identify the wanted element, create a highlight around it, *and then* scroll it into view.